rllib/core/rl_trainer/tests/test_trainer_runner.py

```diff
@@ -197,6 +197,44 @@ def test_add_remove_module(self):
             runner.shutdown()
             del runner
 
+    def test_async_update(self):
+        """Test that async style updates converge to the same result as sync."""
+        fws = ["tf", "torch"]
+        # block=True only needs to be tested for the most complex case.
+        # so we'll only test it for multi-gpu-ddp.
+        scaling_modes = ["multi-gpu-ddp"]
+        test_iterator = itertools.product(fws, scaling_modes)
+
+        for fw, scaling_mode in test_iterator:
+            print(f"Testing framework: {fw}, scaling mode: {scaling_mode}.")
+            env = gym.make("CartPole-v1")
+            scaling_config = REMOTE_SCALING_CONFIGS[scaling_mode]
+            runner = get_trainer_runner(fw, env, scaling_config)
+            reader = get_cartpole_dataset_reader(batch_size=500)
+            min_loss = float("inf")
+            for iter_i in range(1000):
+                batch = reader.next()
+                results = runner.update(
+                    batch.as_multi_agent(), block=False, reduce_fn=None
```

Comment:
How do we test that this line of code is really non-blocking?

Reply:
We could time it and check to see that it takes less time than doing a blocking call?

We could provide an option to return the object futures, that can be used specifically for testing. We would have to be extra careful and tell users to never use it unless for testing.

Reply:
time it. compare blocking vs. non-blocking calls. non-blocking should be substantially faster.

Reply:
done